### PR TITLE
feat(notifier): `hide_from_history` option

### PR DESCRIPTION
**Pull request by kawre**
## Description

Option to exclude some notifications from notification history,
for example, when dealing with spinners.

### CODE REVIEW

File: lua/snacks/notifier.lua [hunk 1]
```diff
@@ -34,6 +34,7 @@ local uv = vim.uv or vim.loop
 ---@field style? snacks.notifier.style
 ---@field opts? fun(notif: snacks.notifier.Notif) -- dynamic opts
 ---@field hl? snacks.notifier.hl -- highlight overrides
+---@field hide_from_history? boolean
```
Comment (folke) [inline lua/snacks/notifier.lua:37]: Can yo change it to `history`? And then check for `opts.history ~= false`?

Comment (kawre) [reply lua/snacks/notifier.lua:37]: nvim-notify already uses `hide_from_history`, should i check for both `history` and `hide_from_history`?

Comment (folke) [reply lua/snacks/notifier.lua:37]: nope, not needed to be compatible with `nvim-notify`

**Comment by folke** [comment]
ty!

**Pull request merged**